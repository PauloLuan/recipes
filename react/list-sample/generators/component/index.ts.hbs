export * from './{{pascalCase componentName}}.component'
