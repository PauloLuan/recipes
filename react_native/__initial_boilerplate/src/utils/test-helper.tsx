import { render, RenderAPI } from '@testing-library/react-native'
import React from 'react'
import { ThemeProvider } from 'styled-components/native'
import theme from '../global/styles/theme'

export const renderWithTheme = (children: React.ReactNode): RenderAPI =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
