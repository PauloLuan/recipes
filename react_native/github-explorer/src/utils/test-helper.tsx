import { render, RenderAPI } from '@testing-library/react-native'
import React from 'react'
import { ThemeProvider } from 'styled-components/native'
import theme from '../global/styles/theme'

export const renderWithTheme = (children: React.ReactNode): RenderAPI =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const mockNavigation = () => {
  const mockedNavigate = jest.fn()

  jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual('@react-navigation/native')
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: mockedNavigate
      })
    }
  })

  return { mockedNavigate }
}
