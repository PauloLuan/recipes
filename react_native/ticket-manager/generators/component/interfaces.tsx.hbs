import { I{{capitalize componentName}}Props } from 'native-base'

export interface {{capitalize componentName}}Props extends I{{capitalize componentName}}Props {
  testId?: string
}
