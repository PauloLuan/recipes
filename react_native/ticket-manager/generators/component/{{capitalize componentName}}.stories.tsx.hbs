import { storiesOf } from '@storybook/react-native'
import React from 'react'
import { {{capitalize componentName}} } from './{{capitalize componentName}}'

storiesOf('Components/{{capitalize componentName}}', module).add('default: ', () => <{{capitalize componentName}} />)
