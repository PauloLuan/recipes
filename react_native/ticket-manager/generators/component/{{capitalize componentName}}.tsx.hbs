import React from 'react'
import { I{{capitalize componentName}}Props, {{capitalize componentName}} as NativeBase{{capitalize componentName}} } from 'native-base'

export interface {{capitalize componentName}}Props extends I{{capitalize componentName}}Props {
  testId?: string
}

export const {{capitalize componentName}} = ({
  testId = '{{kebabCase componentName}}-id',
  ...props
}: {{capitalize componentName}}Props) => {
  return (
    <NativeBase{{capitalize componentName}} testID={testId} {...props}></NativeBase{{capitalize componentName}}>
  )
}
