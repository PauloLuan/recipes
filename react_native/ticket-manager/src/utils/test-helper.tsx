import { render, RenderAPI } from '@testing-library/react-native'
import { NativeBaseProvider } from 'native-base'
import React from 'react'
import { theme } from '../global/styles/theme'

export const renderWithTheme = (children: React.ReactNode): RenderAPI =>
  render(<NativeBaseProvider theme={theme}>{children}</NativeBaseProvider>)
