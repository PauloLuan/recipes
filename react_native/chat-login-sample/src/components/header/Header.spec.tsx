import { waitFor } from '@testing-library/react-native'
import React from 'react'
import { renderWithTheme } from '../../utils/test-helper'
import { Header } from './Header.component'

describe('<Header />', () => {
  const testId = 'component-header-id'

  it('should render the Header component', async () => {
    const { queryByTestId, toJSON } = renderWithTheme(
      <Header testId={testId} />
    )
    await waitFor(() => expect(queryByTestId(testId)).toBeTruthy())
    expect(toJSON()).toMatchSnapshot()
  })
})
