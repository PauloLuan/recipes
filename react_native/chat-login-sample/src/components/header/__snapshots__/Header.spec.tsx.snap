// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the Header component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "space-between",
        "marginBottom": 15,
        "marginLeft": 15,
        "marginRight": 15,
        "marginTop": 15,
      },
    ]
  }
  testID="component-header-id"
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto_500Medium",
          "fontSize": 24,
          "fontWeight": "bold",
        },
      ]
    }
  >
    React Native Chat
  </Text>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "avatarSize": 48,
        "borderRadius": 26,
        "colors": Array [
          4279994175,
          4294954270,
        ],
        "containerSize": 52,
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 26,
          "height": 52,
          "justifyContent": "center",
          "width": 52,
        },
      ]
    }
  >
    <Image
      avatarSize={48}
      borderRadius={26}
      source={
        Object {
          "uri": "file://assets/full/path/to/directory/name.type",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 26,
            "height": 48,
            "width": 48,
          },
        ]
      }
      testID="component-user-photo-id"
    />
  </ViewManagerAdapter_ExpoLinearGradient>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      Sair
    </Text>
  </View>
</View>
`;
